### PR TITLE
[MM-49821] Allow pasting multiple emails into invite modal

### PR DESCRIPTION
**Pull request by kostaspt**
#### Summary
When pasting multiple _valid_ emails separated with ` `, `,`, `;` it will be handled as if a user typed all those and pressed enter afterward.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49821

#### Related Pull Requests
N/A

#### Screenshots
https://user-images.githubusercontent.com/684680/214080247-d9d46c1c-98a5-4d7d-afcd-0f5feb1c076f.mp4

#### Release Note
```release-note
Handle multiple emails at once when inviting users.
```

### CODE REVIEW

**Comment by kostaspt** [comment]
@laneycs Do you think we're okay with this state so we can move on to dev review and QA? Or is there more functionality we want to include at this stage?

**Review by laneycs — APPROVED** [review]
@kostaspt We are ready to roll. Noting that this has been tested and approved by 3 team members in https://community.mattermost.com/private-core/pl/qgoyid7m93fhjbrnfsw4md7d6y

**Comment by mkraft** [comment]
Is this meant to be draft still or ready for us to review?

**Comment by kostaspt** [comment]
> Is this meant to be draft still or ready for us to review?

I had in mind to convert it to regular PR, but I forgot – sorry.

File: components/widgets/inputs/users_emails_input.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ export default class UsersEmailsInput extends React.PureComponent<Props, State>
     }
 
     getOptionValue = (user: UserProfile | EmailInvite): string => {
-        if (user.hasOwnProperty('id')) {
+        if (Object.prototype.hasOwnProperty.call(user, 'id')) {
```
Comment (mkraft) [inline components/widgets/inputs/users_emails_input.tsx:119]: Can we use `if (user instanceof UserProfile) {`?

File: components/widgets/inputs/users_emails_input.tsx [hunk 2]
```diff
@@ -235,41 +235,74 @@ export default class UsersEmailsInput extends React.PureComponent<Props, State>
         NoOptionsMessage: this.props.suppressNoOptionsMessage ? () => null : this.NoOptionsMessage,
         MultiValueRemove: this.MultiValueRemove,
         IndicatorsContainer: () => null,
+        Input: (props: InputProps) => {
+            const handlePaste = (e: ClipboardEvent) => {
+                if (!this.props.emailInvitationsEnabled) {
+                    return;
```
Comment (mkraft) [inline components/widgets/inputs/users_emails_input.tsx:241]: Does this mean we won't be supporting pasting username pasting?

File: components/widgets/inputs/users_emails_input.tsx (hunk 1 above)
Comment (kostaspt) [reply components/widgets/inputs/users_emails_input.tsx]: This specific change throws a `TS2693: 'UserProfile' only refers to a type, but is being used as a value here.`, but I've improved it [a bit for readability](https://github.com/mattermost/mattermost-webapp/pull/12052/commits/d096f04f4a2ad7fb93c84e50c66756607785780e).

File: components/widgets/inputs/users_emails_input.tsx (hunk 2 above)
Comment (kostaspt) [reply components/widgets/inputs/users_emails_input.tsx:245]: I thought we wanted to import bulk emails, not usernames, but re-reading the description, this isn't the case. So I'll refactor to accept usernames as well.

**Comment by kostaspt** [comment]
@BenCookie95 Sounds good. I'll add this part to this PR.

**Comment by BenCookie95** [comment]
> @BenCookie95 Sounds good. I'll add this part to this PR.

Thanks Kostas!

**Comment by kostaspt** [comment]
New cases handled:
* Email pasted that is an existing user is being converted to the actual username
* Email that are existing users and the user is already in the list is omitted
* Usernames that are valid are being added now

**Comment by kostaspt** [comment]
@BenCookie95 Does [this](https://github.com/mattermost/mattermost-webapp/pull/12052/commits/a5370ae4a234d6da84fe651722adf52a2024be9b) work for you? If this prop is triggered more than once, we can assume it's being done by paste. If not, then by type.

**Review by mkraft — APPROVED** [review]
LGTM!

**Comment by BenCookie95** [comment]
> @BenCookie95 Does [this](https://github.com/mattermost/mattermost-webapp/pull/12052/commits/a5370ae4a234d6da84fe651722adf52a2024be9b) work for you? If this prop is triggered more than once, we can assume it's being done by paste. If not, then by type.

That works, thanks Kostas

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — COMMENTED** [review]
Hi @kostaspt, I found two issues. 

1. Other special characters are also getting accepted.
Example: `test1@exapmle.com % test2@exapmle.com `

2. The email of the Deactivated user is getting accepted. Please see the below.

https://user-images.githubusercontent.com/6909890/217361268-cabe508f-1657-4b86-b90d-e58e8c017975.mp4

**Comment by kostaspt** [comment]
1. Not sure if this is an issue, as the `%` it's just being discarded. It's actually 3 tokens separated by a space, so: `test1@exapmle.com`, `%`, `test2@exapmle.com ` -> valid, discarded, valid
But maybe we should handle this some other way, e.g., have a red chip with the "discarded" items cc @annapohorielova111 

2. Your right, I'll look into it. It's related to how I did the search for the username. (It's funny how the [MM-50229](https://mattermost.atlassian.net/browse/MM-50229) frustrates you as well 😛)

**Comment by kostaspt** [comment]
@furqanmlk Not sure if this is better, but beyond that is potentially a different UX direction.

https://user-images.githubusercontent.com/684680/217549761-4d23345c-09a9-4442-8ce1-9c7e906fc618.mp4

**Comment by furqanmlk** [comment]
@kostaspt 
Can you please fix the conflict ?

**Review by furqanmlk — APPROVED** [review]
Working as expected

**Pull request merged**